lib/logstash/inputs/http.rb

```diff
@@ -75,11 +83,20 @@ def register
   end # def register
 
   def run(queue)
```

Comment:
Is there any reason why we don't use rack middleware for this kind of authorization?

We are already using the rack api for all the answers I think it would be more flexible to rely on rack for this kind of behaviors. See https://github.com/rack/rack/blob/master/lib/rack/auth/basic.rb, The library is well tested and maintained.

Also we could probably support multipart post with it?

What do you think?